### PR TITLE
Run Docker/runasnonroot

### PR DESCRIPTION
**Pull request by usma0118**
from discussion: #657 

- Modified to run docker image as user mealie (PUID: 911, PGID: 911)
- Changed caddy port from 80 to 9080 for non-root binding (breaking change for existing deployments)
- removed groupmod from run script and handled in docker file.
- Bumped image to python:3.9.6
- moved scheduler.db to /app/temp as no exposed volume, Thanks to @articuno1au
- bumped poetry version.

I have tested this on docker and kubernetes (via helm).

### CODE REVIEW

File: Caddyfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	admin off
 }
 
-:80 {
+:9080 {
```
Comment (hay-kot) [inline Caddyfile:6]: Can you set this as an ENV variable that defaults to 80? That way we can avoid the breaking change but still allow others to configure as necessary. Once we move to v1 I can make the port change by default.

https://caddyserver.com/docs/caddyfile-tutorial#environment-variables

File: mealie/run.sh [hunk 2]
```diff
@@ -5,20 +5,6 @@ set -e
...
-add_user() {
-    groupmod -o -g "$PGID" abc
-    usermod -o -u "$PUID" abc
-
-    echo "
-    User uid:    $(id -u abc)
-    User gid:    $(id -g abc)
-    "
-    chown -R abc:abc /app
-}
```
Comment (hay-kot) [inline mealie/run.sh:21]: Why isn't this needed anymore? Don't we still need to set permissions on the mounted directories?

**Review by hay-kot — CHANGES_REQUESTED** [review]
Overall I think it looks good. You do need to set the PR to work from the dev branch as is described in the[ contributions guide ](https://hay-kot.github.io/mealie/contributors/developers-guide/code-contributions/)

Once those are done, I'll pull it down and do a thorough testing. Thanks for the work on this!

File: Caddyfile (hunk 1 above)
Comment (usma0118) [reply Caddyfile:6]: good idea, will do.

**Pull request closed without merging**